cmd/soroban-cli/src/commands/init.rs

```diff
@@ -0,0 +1,329 @@
+
+    // if there are with-contract flags, include the example contracts
+    if include_example_contracts(with_contracts) {
+        println!("Including example contracts: {with_contracts:?}");
+
+        // create an examples temp dir in the temp dir
+        let examples_dir = tempfile::tempdir()?;
+
+        // clone the soroban-examples repo into temp dir
+        clone_repo(SOROBAN_EXAMPLES_URL, examples_dir.path())?;
+
+        // copy the example contracts into the project
+        copy_example_contracts(examples_dir.path(), project_path, with_contracts)?;
+    }
+
+    Ok(())
+}
+
+fn copy_example_contracts(
+    from: &Path,
+    to: &Path,
+    contracts: &[ExampleContract],
+) -> Result<(), Error> {
+    let project_contracts_path = to.join("contracts");
+    for contract in contracts {
+        let contract_as_string = contract.to_string();
+        let contract_path = Path::new(&contract_as_string);
+        let from_contract_path = from.join(contract_path);
+        let to_contract_path = project_contracts_path.join(contract_path);
+        std::fs::create_dir_all(&to_contract_path)?;
+
+        copy_contents(&from_contract_path, &to_contract_path)?;
+        edit_contract_cargo_file(&to_contract_path)?;
+    }
+
+    Ok(())
+}
+
+fn edit_contract_cargo_file(contract_path: &Path) -> Result<(), Error> {
+    let cargo_path = contract_path.join("Cargo.toml");
+    let cargo_toml_str = read_to_string(&cargo_path)?;
+    let mut doc = cargo_toml_str.parse::<Document>().unwrap();
+
+    let mut workspace_table = InlineTable::new();
+    workspace_table.insert("workspace", Value::Boolean(Formatted::new(true)));
+
+    doc["dependencies"]["soroban-sdk"] =
+        toml_edit::Item::Value(Value::InlineTable(workspace_table.clone()));
+    doc["dev_dependencies"]["soroban-sdk"] =
+        toml_edit::Item::Value(Value::InlineTable(workspace_table));
+
+    doc.remove("profile");
+
+    std::fs::write(&cargo_path, doc.to_string())?;
+
+    Ok(())
+}
+
+fn include_example_contracts(contracts: &[ExampleContract]) -> bool {
+    !(contracts.len() == 1 && contracts[0] == ExampleContract::None)
+}
+
+fn clone_repo(from_url: &str, to_path: &Path) -> Result<(), Error> {
+    let mut fetch = gix::clone::PrepareFetch::new(
+        from_url,
+        to_path,
+        gix::create::Kind::WithWorktree,
+        gix::create::Options {
+            destination_must_be_empty: false,
+            fs_capabilities: None,
+        },
+        gix::open::Options::isolated(),
+    )
+    .map_err(Box::new)?
+    .with_shallow(gix::remote::fetch::Shallow::DepthAtRemote(
+        NonZeroU32::new(1).unwrap(),
+    ));
+
+    let (mut prepare, _outcome) = fetch
+        .fetch_then_checkout(gix::progress::Discard, &AtomicBool::new(false))
+        .map_err(Box::new)?;
+
+    let (_repo, _outcome) =
+        prepare.main_worktree(gix::progress::Discard, &AtomicBool::new(false))?;
+
+    Ok(())
+}
+
+fn copy_contents(from: &Path, to: &Path) -> Result<(), Error> {
```

Comment:
Could we log to console each file and directory being written? If we skip files that already exist we can indicate so in the output.

Reply:
e.g.:

```
❯ soroban init . --with-contract hello-world --with-contract custom-types
ℹ️ Initializing project at .
ℹ️ Skipped creating .gitignore as it already exists
ℹ️ Skipped creating Cargo.toml as it already exists
ℹ️ Skipped creating Cargo.lock as it already exists
ℹ️ Skipped creating README.md as it already exists
ℹ️ Initializing example contract: hello_world
➕ Writing contracts/hello_world/Cargo.toml
➕ Writing contracts/hello_world/src/lib.rs
➕ Writing contracts/hello_world/src/test.rs
ℹ️ Initializing example contract: custom_types
ℹ️ Skipped creating contracts/custom_types as it already exists
✅ Project initialized
```

Reply:
Yeah, great idea!

Reply:
updated in [31fdbf9](https://github.com/stellar/soroban-tools/pull/1156/commits/31fdbf999f092fd0d49d14fc812ca4a4635aabe1)